File: AtlasWeb/Core/Modules/Kaynak_Tanimlari/Arsiv_Tanimlari/ArchiveRoomsDefFormViewModel.ts
//$0E88E0C0
import { BaseViewModel } from "app/NebulaClient/Model/BaseViewModel";
import { ResArchiveRoom, ResCabinet, ResShelf } from "NebulaClient/Model/AtlasClientModel";
import { ResFloor } from "NebulaClient/Model/AtlasClientModel";
import { ResBuildingWing } from "NebulaClient/Model/AtlasClientModel";
import { ResBuilding } from "NebulaClient/Model/AtlasClientModel";
import { Guid } from "app/NebulaClient/Mscorlib/Guid";
import { Type } from "app/NebulaClient/ClassTransformer";

export class ArchiveRoomsDefFormViewModel extends BaseViewModel {
    @Type(() => ResArchiveRoom)
    public _ResArchiveRoom: ResArchiveRoom = new ResArchiveRoom();
    @Type(() => ResFloor)
    public ResFloors: Array<ResFloor> = new Array<ResFloor>();
    @Type(() => ResBuildingWing)
    public ResBuildingWings: Array<ResBuildingWing> = new Array<ResBuildingWing>();
    @Type(() => ResBuilding)
    public ResBuildings: Array<ResBuilding> = new Array<ResBuilding>();
    @Type(() => ResCabinet)
    public CabinetList: Array<ResCabinet> = new Array<ResCabinet>();
    @Type(() => RoomModel)
    public RoomList: Array<RoomModel> = new Array<RoomModel>();
    @Type(() => ResShelf)
    public ShelfList: Array<ResShelf> = new Array<ResShelf>();
    @Type(() => CabinetShelf)
    public CabinetShelf: Array<CabinetShelf> = new Array<CabinetShelf>();

}

export class RoomModel {
    public ObjectID: Guid;
    public RoomName: string;
    public BuildingName: string;
    public WingName: string;
    public FloorName: string;
    public Active: boolean;
    public Room: ResArchiveRoom;
}

export class CabinetShelf{
    public ShelfObjectID: Guid;
    public CabinetObjectID: Guid;
}

File: AtlasWeb/Core/Modules/Kaynak_Tanimlari/Arsiv_Tanimlari/ArchiveRoomsDefFormViewModel.cs
//$AE758851
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;

namespace Core.Controllers
{
    public partial class ResArchiveRoomServiceController
    {
        partial void PreScript_ArchiveRoomsDefForm(ArchiveRoomsDefFormViewModel viewModel, ResArchiveRoom resArchiveRoom, TTObjectContext objectContext)
        {

            viewModel.RoomList = new List<RoomModel>();
            List<ResArchiveRoom> roomList = ResArchiveRoom.GetArchiveRoomListDef(objectContext, "").ToList();
            foreach (ResArchiveRoom item in roomList)
            {
                RoomModel model = new RoomModel();
                model.ObjectID = item.ObjectID;
                model.RoomName = item.Name;
                model.BuildingName = item.ResBuilding.Name;
                if (item.ResBuildingWing != null)
                    model.WingName = item.ResBuildingWing.Name;
                if (item.ResFloor != null)
                    model.FloorName = item.ResFloor.Name;
                model.Active = item.IsActive;
                model.Room = item;
                viewModel.RoomList.Add(model);
            }

            viewModel.CabinetList = resArchiveRoom.ResCabinets.ToList();
            viewModel.ShelfList = new List<ResShelf>();

            foreach (var item in viewModel.CabinetList)
            {
                viewModel.ShelfList = viewModel.ShelfList.Concat(item.ResShelves).ToList();
            }
        }

    }
}

namespace Core.Models
{
    public class RoomModel
    {
        public Guid ObjectID { get; set; }
        public string RoomName { get; set; }
        public string BuildingName { get; set; }
        public string WingName { get; set; }
        public string FloorName { get; set; }
        public bool? Active { get; set; }
        public ResArchiveRoom Room { get; set; }
    }
    public partial class ArchiveRoomsDefFormViewModel : BaseViewModel
    {
        public List <RoomModel> RoomList { get; set; }
        public List<ResCabinet> CabinetList { get; set; }
        public List<ResShelf> ShelfList { get; set; }
        public List<CabinetShelf> CabinetShelf { get; set; }


    }
    public class CabinetShelf
    {
        public Guid CabinetObjectID { get; set; }
        public Guid ShelfObjectID { get; set; }
    }
}
